src/configManager.js

```diff
@@ -0,0 +1,73 @@
+import * as LogManager from 'aurelia-logging';
+import extend from 'extend';
+import {GlobalConfig} from './globalConfig';
+
+export const GLOBAL_CONFIG_ID = 'global-config';
+
+/**
+ * The ConfigManager class
+ * Register and retrieve configs
+ */
+export class ConfigManager {
+  /**
+   * POJO map of the registered configs
+   * @type {Object}
+   */
+  configs = {};
+
+  /**
+   * Configuration function
+   * @param  {[Object]}       incomming The configuration object
+   * @return {Promise}
+   */
+  configure(incomming = {plugins: []}) {
```

Comment:
The way it is now we have a `configure` method and the configs object. The configure method extends the configs object. Reading and writing the configs is as simple as traversing threw the configs object. This is simple and self explanatory. I would keep it as is. I would however discourage users from reading and writing to this object directly. Just like the BaseConfig has methods for altering the config object, the configManager could have something similar for getting, setting or removing configs.